src/service/subscription.ts

```diff
@@ -42,6 +42,14 @@ export default class SubscriptionService {
     //
   }
 
+  /**
+   *   If passed, it would serve as the default error handler at SubscriptionService level
+   */
+  public static handleError(error: Error): void {
+    // default error handling logic
+    Logger.Instance.error({ error }, 'Received Unexpected Error');
+  }
```

Comment:
Let's rethrow the error in the default handler, so application can handle it on their own
```suggestion
  public static handleError(error: Error): void {
    // default error handling logic
    Logger.Instance.error({ error }, 'Received Unexpected Error');
    throw error;
  }
```